File: java/core/src/main/java/com/github/fernthedev/core/packets/HashedPasswordPacket.java
package com.github.fernthedev.core.packets;

import com.github.fernthedev.core.data.HashedPassword;
import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
@PacketInfo(name = "HASHED_PASSWORD_PACKET")
public class HashedPasswordPacket extends Packet {

    private HashedPassword hashedPassword;

    public HashedPasswordPacket(String password) {
        hashedPassword = new HashedPassword(password);
    }

}

File: java/core/src/main/java/com/github/fernthedev/lightchat/core/packets/HashedPasswordPacket.kt
package com.github.fernthedev.lightchat.core.packets

import com.github.fernthedev.lightchat.core.data.HashedPassword
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
@PacketInfo(name = "HASHED_PASSWORD_PACKET")
class HashedPasswordPacket(
    var hashedPassword: HashedPassword
) : PacketJSON() {

    constructor(password: String) : this(HashedPassword(password))
}